src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml

```diff
@@ -272,6 +272,77 @@
         </foreach>
     </sql>
 
+    <sql id="applyNumericalDataFilter">
+        <foreach item="dataFilterValue" collection="${data_filter_collection}" open="((" separator=") OR (" close="))">
+            <trim prefix="" prefixOverrides="AND">
+                <!-- Special case: NA -->
+                <if test="dataFilterValue.value eq 'NA'">
+                    AND
+                    <include refid="isAttributeValueNA">
+                        <property name="attribute_value" value="${attribute_value}"/>
+                    </include>
+                </if>
+                <!--
+                    Special case: Non-numerical value within numerical filter collection.
+                    Even if the data type is numerical we can still have non-numerical values
+                    such as "UNKNOWN", "NOT COLLECTED", "NOT RELEASED", etc.
+                -->
+                <if test="dataFilterValue.value != null">
+                    AND (
+                    <include refid="normalizeAttributeValue">
```

Comment:
I wouldn't do the normalizeAttributeValue. You are just testing multiple values of NA?

Reply:
`normalizeAttributeValue` used to normalize values other than `NA` but we got rid of that part at some point by upper casing everything. So I guess normalizing here is redundant now.

We are still doing this in multiple places where we filter categorical data. For example, see https://github.com/cBioPortal/cbioportal/blob/d9459f39a92ab251178793349e1b9ba003148238/src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml#L483-L489
and
https://github.com/cBioPortal/cbioportal/blob/d9459f39a92ab251178793349e1b9ba003148238/src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml#L614-L620

These are also most likely redundant now, and I think we should address this duplication as well, but probably in a separate PR.

Reply:
Removed `normalizeAttributeValue` from all these 3 places. We may still want to address the code duplication for categorical filters at some point.